org.osgi.test.junit4/src/main/java/org/osgi/test/common/annotation/Property.java

```diff
@@ -105,4 +120,22 @@ public enum ValueSource {
 	 */
 	ValueSource source() default ValueSource.Value;
 
+	/**
+	 * If any template arguments are set then the resolved value of this
+	 * {@link Property} annotation will be used as a template in
+	 * {@link String#format} with the resolved {@link ValueArgument} values used
+	 * as arguments.
+	 * <p>
+	 * Note that any defaulting or processing according to the {@link #source()}
+	 * will happen <em>before</em> the formatting is applied, so template
+	 * arguments cannot be used, for example, to change the name of a system
+	 * property. Conversely, the scalar conversion will happen <em>after</em>
+	 * the template is applied, meaning that numeric values can be assembled
+	 * from multiple template arguments.
+	 *
+	 * @return the arguments that should be used with the template
+	 * @since 1.2
+	 */
+	ValueArgument[] templateArguments() default {};
```

Comment:
There is a mismatch here between the element name (template) and the element value type (value).